File: exercism/go/rna-transcription/rna_transcription.go
package strand

import "bytes"

// DnaToRna maps DNA nucleotides to RNA nucleotides
var DnaToRna = map[rune]rune{
	'C': 'G',
	'G': 'C',
	'T': 'A',
	'A': 'U',
}

// ToRNA transcribes a DNA strand to its RNA complement
func ToRNA(dna string) string {
	var buf bytes.Buffer

	for _, nuc := range dna {
		buf.WriteRune(DnaToRna[nuc])
	}

	return buf.String()
}

File: exercism/c/rna-transcription/src/rna_transcription.c
#include "rna_transcription.h"
#include <string.h>
#include <stdlib.h>

char *to_rna(const char *dna) {
  int i, len = strlen(dna);
  char *rna = (char *)malloc(len * sizeof(char));
  if (rna == NULL)
    return NULL;

  for (i = 0; i < len; ++i) {
    switch (*(dna+i)) {
      case 'G':
        *(rna+i) = 'C';
        break;
      case 'C':
        *(rna+i) = 'G';
        break;
      case 'T':
        *(rna+i) = 'A';
        break;
      case 'A':
        *(rna+i) = 'U';
        break;
      default:
        free(rna);
        return NULL;
    }
  }

  *(rna+i) = '\0';

  return rna;
}
